xcube_cds/store.py

```diff
@@ -749,11 +772,26 @@ def open_data(self, data_id: str, opener_id: Optional[str] = None,
     # Implementation helpers
 
     @staticmethod
-    def _assert_valid_type_id(type_id):
-        if type_id is not None and type_id != TYPE_ID_DATASET:
+    def _validate_type_specifier(type_specifier: Union[str, TypeSpecifier]):
+        if not CDSDataStore._is_type_specifier_satisfied(type_specifier):
             raise DataStoreError(
-                f'Data type identifier must be "{TYPE_ID_DATASET}", '
-                f'but got "{type_id}"')
+                f'Supplied type specifier "{type_specifier}" is not compatible '
+                f'with "{TYPE_SPECIFIER_CUBE}."'
+            )
+
+    @staticmethod
+    def _is_type_specifier_satisfied(
+            type_specifier: Union[str, TypeSpecifier]
+            ) -> bool:
```

Comment:
Weird function head formatting:

```suggestion
    def _is_type_specifier_satisfied(type_specifier: Union[str, TypeSpecifier]) -> bool:
```

xcube style:

If there is enough space, put all params in one line.
If not,
- first param directly behind first paren,
- next params aligned below first
- last paren directly behind last param

Reply:
All parameters (and type specifiers) on one line makes for 88 characters, exceeding PEP-8's 79-column limit. Even with the return type on the next line, it goes over. And when there's only one parameter, the three bullet points simplify to "all params on one line". But I can at least remove the line break between the parameter and the closing parenthesis.